### ISSUE TITLE
If getting page sources fails because of driver issues, test is not included in BDD report

### ISSUE THREAD

**Issue by kutzi** [serenity-bdd/serenity-core#1247]
We've seen something similar to https://github.com/serenity-bdd/serenity-core/issues/1108, but related to page sources:

```
13:13:44 (Id: 8) [main] ERROR SerenityMain - Exception in Cucumber runtime
13:13:44 org.openqa.selenium.TimeoutException: timeout
13:13:44   (Session info: chrome=63.0.3239.132)
13:13:44   (Driver info: chromedriver=2.34.522913 (36222509aa6e819815938cbf2709b4849735537c),platform=Linux 4.4.119-1.el7.elrepo.x86_64 x86_64) (WARNING: The server did not provide any stacktrace information)
13:13:44 Command duration or timeout: 0 milliseconds
13:13:44 Build info: version: '3.11.0', revision: 'e59cfb3', time: '2018-03-11T20:26:55.152Z'
13:13:44 System info: host: 'xxx', ip: 'xxx', os.name: 'Linux', os.arch: 'amd64', os.version: '4.4.119-1.el7.elrepo.x86_64', java.version: '1.8.0_151'
13:13:44 Driver info: org.openqa.selenium.chrome.ChromeDriver
13:13:44 Capabilities {acceptSslCerts: true, applicationCacheEnabled: false, browserConnectionEnabled: false, browserName: chrome, chrome: {chromedriverVersion: 2.34.522913 (36222509aa6e81..., userDataDir: /tmp/.org.chromium.Chromium...}, cssSelectorsEnabled: true, databaseEnabled: false, handlesAlerts: true, hasTouchScreen: false, javascriptEnabled: true, locationContextEnabled: true, mobileEmulationEnabled: false, nativeEvents: true, networkConnectionEnabled: false, pageLoadStrategy: normal, platform: LINUX, platformName: LINUX, rotatable: false, setWindowRect: true, takesHeapSnapshot: true, takesScreenshot: true, unexpectedAlertBehaviour: accept, unhandledPromptBehavior: accept, version: 63.0.3239.132, webStorageEnabled: true}
13:13:44 Session ID: 8e4408bbae5a813c91bbaaf873d5a7ea
13:13:44 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
13:13:44 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
13:13:44 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
13:13:44 	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
13:13:44 	at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:214)
13:13:44 	at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:166)
13:13:44 	at org.openqa.selenium.remote.http.JsonHttpResponseCodec.reconstructValue(JsonHttpResponseCodec.java:40)
13:13:44 	at org.openqa.selenium.remote.http.AbstractHttpResponseCodec.decode(AbstractHttpResponseCodec.java:80)
13:13:44 	at org.openqa.selenium.remote.http.AbstractHttpResponseCodec.decode(AbstractHttpResponseCodec.java:44)
13:13:44 	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:158)
13:13:44 	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:83)
13:13:44 	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:545)
13:13:44 	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:602)
13:13:44 	at org.openqa.selenium.remote.RemoteWebDriver.getPageSource(RemoteWebDriver.java:431)
13:13:44 	at net.serenitybdd.core.photography.PageSourceRecorder.getPageSource(PageSourceRecorder.java:40)
13:13:44 	at net.serenitybdd.core.photography.PageSourceRecorder.intoDirectory(PageSourceRecorder.java:26)
13:13:44 	at net.thucydides.core.steps.BaseStepListener.grabScreenshot(BaseStepListener.java:865)
13:13:44 	at net.thucydides.core.steps.BaseStepListener.take(BaseStepListener.java:766)
13:13:44 	at net.thucydides.core.steps.BaseStepListener.takeEndOfStepScreenshotFor(BaseStepListener.java:747)
13:13:44 	at net.thucydides.core.steps.BaseStepListener.stepFailed(BaseStepListener.java:659)
13:13:44 	at net.thucydides.core.steps.StepEventBus.stepFailed(StepEventBus.java:392)
13:13:44 	at cucumber.runtime.formatter.SerenityReporter.failed(SerenityReporter.java:778)
13:13:44 	at cucumber.runtime.formatter.SerenityReporter.recordStepResult(SerenityReporter.java:731)
13:13:44 	at cucumber.runtime.formatter.SerenityReporter.handleResult(SerenityReporter.java:721)
13:13:44 	at cucumber.runtime.formatter.SerenityReporter.handleTestStepFinished(SerenityReporter.java:291)
13:13:44 	at cucumber.runtime.formatter.SerenityReporter.lambda$new$4(SerenityReporter.java:115)
13:13:44 	at cucumber.runner.EventBus.send(EventBus.java:28)
13:13:44 	at cucumber.api.TestStep.run(TestStep.java:90)
13:13:44 	at cucumber.api.TestCase.run(TestCase.java:58)
13:13:44 	at cucumber.runner.Runner.runPickle(Runner.java:80)
13:13:44 	at cucumber.runtime.Runtime.runFeature(Runtime.java:119)
13:13:44 	at cucumber.runtime.Runtime.run(Runtime.java:104)
13:13:44 	at cucumber.runtime.Runtime$run.call(Unknown Source)
13:13:44 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
13:13:44 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
13:13:44 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:120)
13:13:44 	at SerenityMain.run(SerenityMain.groovy:41)
13:13:44 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
13:13:44 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
13:13:44 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
13:13:44 	at java.lang.reflect.Method.invoke(Method.java:498)
13:13:44 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:98)
13:13:44 	at org.codehaus.groovy.runtime.callsite.StaticMetaMethodSite$StaticMetaMethodSiteNoUnwrapNoCoerce.invoke(StaticMetaMethodSite.java:151)
13:13:44 	at org.codehaus.groovy.runtime.callsite.StaticMetaMethodSite.callStatic(StaticMetaMethodSite.java:102)
13:13:44 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:55)
13:13:44 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:197)
13:13:44 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:217)
13:13:44 	at SerenityMain.main(SerenityMain.groovy:26)
```

**Comment by kutzi** [comment]
See https://github.com/serenity-bdd/serenity-core/pull/1248

**Comment by wakaleo** [comment]
Should be fixed in 1.9.17

**Comment by kutzi** [comment]
Thanks
